File: CleverTapSDK/CleverTapTrackedViewController.m

#import "CleverTapTrackedViewController.h"
#import "CleverTap.h"

@implementation CleverTapTrackedViewController

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    if (self.screenName) {
        [[CleverTap sharedInstance] recordScreenView:self.screenName];
    }
}


@end

File: CleverTapSDK/CleverTapTrackedViewController.swift
//
//  CleverTapTrackedViewController.swift
//  CleverTapSDK
//

import UIKit

@objc(CleverTapTrackedViewController)
@objcMembers
open class CleverTapTrackedViewController: UIViewController {
    public var screenName: String?

    open override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        if let screenName = screenName {
            CleverTap.sharedInstance()?.recordScreenView(screenName)
        }
    }
}
